core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -350,7 +350,8 @@ init_frame_refs(uint8 *frame_ref, uint32 cell_num, WASMFunctionInstance *func)
 
 #if UINTPTR_MAX == UINT64_MAX
 #define SET_FRAME_REF(off) *FRAME_REF(off) = *FRAME_REF(off + 1) = 1
-#define CLEAR_FRAME_REF(off) *FRAME_REF(off) = *FRAME_REF(off + 1) = 0
+#define CLEAR_FRAME_REF(off) \
+    off >= local_cell_num ? (*FRAME_REF(off) = *FRAME_REF(off + 1) = 0) : 0
```

Comment:
Should be `if (off >= local_cell_num) (*FRAME_REF(off) = *FRAME_REF(off + 1) = 0)`? Do nothing when `off < local_cell_num`?

Reply:
Here `CLEAR_FRAME_REF` must be an expression, if is a statement and will cause compilation error
When off < local_cell_num, it's the local slot, and it should always be marked as ref, otherwise the object may be freed unexpectly

Reply:
how about
```C
do {
    if (off >= local_cell_num)
        *FRAME_REF(off) = *FRAME_REF(off + 1) = 0;
} while (0)
```

Reply:
`CLEAR_FRAME_REF` is used here:

``` C
#define POP_REF()                                        \
    (opnd_off = GET_OFFSET(), CLEAR_FRAME_REF(opnd_off), \
     GET_REF_FROM_ADDR(frame_lp + opnd_off))
```

It require an expression rather than statement, `if`, `do... while` are statements

Reply:
Got it, had better `: (void)0` instead of `: 0`